data_steward/cdr_cleaner/cleaning_rules/free_text_survey_response_suppression.py

```diff
@@ -0,0 +1,109 @@
+"""
+To ensure participant privacy, remove any records containing concepts related to free text responses
+
+Original Issue: DC-1387
+"""
+
+# Python imports
+import logging
+
+from google.cloud.exceptions import GoogleCloudError
+
+# Project imports
+from utils import pipeline_logging
+from common import JINJA_ENV, OBSERVATION
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.deid.concept_suppression import AbstractBqLookupTableConceptSuppression
+
+LOGGER = logging.getLogger(__name__)
+
+SUPPRESSION_RULE_CONCEPT_TABLE = 'free_text_suppression_concept'
+
+FREE_TEXT_CONCEPT_QUERY = JINJA_ENV.from_string("""
+-- This query generates a lookup table that contains all the suppressed concepts relating to  --
+-- free text generated using REGEX --
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_dataset}}.{{concept_suppression_table}}` AS
+(SELECT * FROM `{{project_id}}.{{dataset_id}}.concept`
+WHERE REGEXP_CONTAINS(concept_code, r'(FreeText)|(TextBox)') OR concept_code = 'notes')
+""")
+
+
+class FreeTextSurveyResponseSuppression(AbstractBqLookupTableConceptSuppression
+                                       ):
+    """
+    Any record in the observation table with a free text concept should be sandboxed and suppressed
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Sandbox and record suppress any records containing concepts related to free text responses'
+        super().__init__(
+            issue_numbers=['DC1387'],
+            description=desc,
+            affected_datasets=[cdr_consts.CONTROLLED_TIER_DEID],
+            affected_tables=[OBSERVATION],
+            project_id=project_id,
+            dataset_id=dataset_id,
+            sandbox_dataset_id=sandbox_dataset_id,
+            concept_suppression_lookup_table=SUPPRESSION_RULE_CONCEPT_TABLE)
+
+    def create_suppression_lookup_table(self, client):
+        """
+
+        :param client:
+        :return:
+
+        raises google.cloud.exceptions.GoogleCloudError if a QueryJob fails
+        """
+        concept_suppression_lookup_query = FREE_TEXT_CONCEPT_QUERY.render(
+            project_id=self.project_id,
+            dataset_id=self.dataset_id,
+            sandbox_dataset=self.sandbox_dataset_id,
+            concept_suppression_table=self.concept_suppression_lookup_table)
+        query_job = client.query(concept_suppression_lookup_query)
+        result = query_job.result()
+
+        if hasattr(result, 'errors') and result.errors:
```

Comment:
When shortened, I got the error:
`AttributeError: '_EmptyRowIterator' object has no attribute 'errors'`

Reply:
@dcampbell-vumc I see what's happening now.  You're looking for errors from the result object instead of from the query_job object.  The row iterator (result object) will never have an errors attribute.  The `errors` attribute isn't defined in the [RowIterator docs](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.table.RowIterator.html#google.cloud.bigquery.table.RowIterator).  What you should be checking for errors is the QueryJob object.

```suggestion
    if query_job.errors or query_job.error_result:
```

Both attributes are available to the query_job object.  [QueryJob docs](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.QueryJob.html#google.cloud.bigquery.job.QueryJob.error_result).

For full disclosure, I've used `query_job.exceptions()` and `query_job.errors`.  I can't remember using `query_job.error_result`.

Reply:
I see that you copied that from the `concept_suppression.py` example.  I'm going to open a bug ticket to fix that one.  I missed it during my review.

Reply:
Once this line of code is updated, I will approve it.